### DISCUSSION TITLE
How to define the child class from the abstract class ghidra.graph.GEdge in python

### DISCUSSION THREAD

**Discussion by shinibufa** [NationalSecurityAgency/ghidra#5805] [Q&A] [answered]
Hello, Ghidra developers. I'm currently working on defining a custom class derived from the abstract class ghidra.graph.GEdge. Here's the code snippet I've been using:

`
class myEdge(GEdge):
    
    def __init__(self, start, end):
        self.start = start
        self.end = end

    def getStart(self):
        return self.start

    def getEnd(self):
        return self.end
`

However, I'm encountering the following error:
`  File "C:\Users\DELL\ghidra_scripts\dominator.py", line 40, in __init__
    self.start = start
AttributeError: read-only attr: start`
Is there a solution to this problem? Thank you for your assistance."

**Comment by Junaid0067** [comment, marked as answer]
You can use properties in Python for this type of tasks. First, you make your attribute 'private' by adding two underscores, then you create a getter method with the @property decorator:

    def __init__(self, start, end):
    self.__start = start
    self.__end = end

    @property
    def start(self):
        return self.__start

    @property
    def end(self):
        return self.__end

**Reply by shinibufa** [reply]
Thank you very much
